dbms/src/DataStreams/WindowTransformAction.h

```diff
@@ -239,7 +311,8 @@ struct WindowTransformAction
     // Auxiliary variable for range frame type when calculating frame_end
     RowNumber prev_frame_end;
 
-    //TODO: used as template parameters
-    bool only_have_row_number = false;
+    std::unique_ptr<Arena> arena;
```

Comment:
The arena is shared by all the aggregation functions, I have some concerns that arena's memory usage will not be properly released for each row

Reply:
> The arena is shared by all the aggregation functions, I have some concerns that arena's memory usage will not be properly released for each row

arena is deleted now